layouts/shortcodes/feature-state.html

```diff
@@ -2,10 +2,56 @@
 {{ $state := .Get "state" }}
 {{ $for_k8s_version := .Get "for_k8s_version" | default (.Page.Param "version")}}
 {{ $is_valid := strings.Contains $valid_states $state }}
-{{ if not $is_valid }}
-{{ errorf "%q is not a valid feature-state, use one of %q" $state $valid_states }}
-{{ else }}
-<div style="margin-top: 10px; margin-bottom: 10px;">
-  <b>{{ T "feature_state" }}</b> <code>Kubernetes {{ $for_k8s_version }} [{{ $state }}]</code>
-</div>
+{{ $feature_gate_name := .Get "feature_gate_name" }}
+
+ <!-- When 'feature_gate_name' parameter is not specified, continue with the values provided. -->
+{{ if not $feature_gate_name }}
+  {{ if not $is_valid }}
+    {{ errorf "%q is not a valid feature-state, use one of %q" $state $valid_states }}
+  {{ else }}
+    <div class="my-2">
+      <b>{{ T "feature_state" }}</b> <code>{{T "feature_state_kubernetes_label" }} {{ $for_k8s_version }} [{{ $state }}]</code>
+    </div>
+  {{ end }}
+
+{{- else -}}
+  <!-- When 'feature_gate_name' is specified, the following code block extracts the status of the feature gate from the description file. -->
+
+  <!-- Set the path for feature gate description files -->
+  {{- $featureGateDescriptionFilesPath := "/docs/reference/command-line-tools-reference/feature-gates" -}}
+
+  <!-- Access the 'English' site and get the feature gate description pages -->
+  {{- with index (where .Site.Sites "Language.Lang" "eq" "en") 0 -}}
```

Comment:
This logic is added to ensure that the feature state is fetched from the English files. It could also be incorporated into the `feature-gate-table` shortcode.